File: backend/Business/LeagueSettingsService.cs
using Microsoft.EntityFrameworkCore;

namespace GolfLeagueManager
{
    public class LeagueSettingsService
    {
        private readonly AppDbContext _context;

        public LeagueSettingsService(AppDbContext context)
        {
            _context = context;
        }

        /// <summary>
        /// Get league settings for a season, creating default settings if none exist
        /// </summary>
        public async Task<LeagueSettings> GetLeagueSettingsAsync(Guid seasonId)
        {
            var settings = await _context.LeagueSettings
                .FirstOrDefaultAsync(ls => ls.SeasonId == seasonId);

            if (settings == null)
            {
                // Create default settings for the season
                settings = CreateDefaultSettings(seasonId);
                _context.LeagueSettings.Add(settings);
                await _context.SaveChangesAsync();
            }

            return settings;
        }

        /// <summary>
        /// Update league settings for a season
        /// </summary>
        public async Task<LeagueSettings> UpdateLeagueSettingsAsync(LeagueSettings settings)
        {
            var existingSettings = await _context.LeagueSettings
                .FirstOrDefaultAsync(ls => ls.SeasonId == settings.SeasonId);

            if (existingSettings == null)
            {
                // Create new settings
                settings.Id = Guid.NewGuid();
                settings.CreatedDate = DateTime.UtcNow;
                _context.LeagueSettings.Add(settings);
            }
            else
            {
                // Update existing settings
                existingSettings.HandicapMethod = settings.HandicapMethod;
                existingSettings.CoursePar = settings.CoursePar;
                existingSettings.CourseRating = settings.CourseRating;
                existingSettings.SlopeRating = settings.SlopeRating;
                existingSettings.MaxRoundsForHandicap = settings.MaxRoundsForHandicap;
                existingSettings.ScoringMethod = settings.ScoringMethod;
                existingSettings.PointsSystem = settings.PointsSystem;
                existingSettings.HoleWinPoints = settings.HoleWinPoints;
                existingSettings.HoleHalvePoints = settings.HoleHalvePoints;
                existingSettings.MatchWinBonus = settings.MatchWinBonus;
                existingSettings.MatchTiePoints = settings.MatchTiePoints;
                existingSettings.UseSessionHandicaps = settings.UseSessionHandicaps;
                existingSettings.AllowHandicapUpdates = settings.AllowHandicapUpdates;
                existingSettings.CustomRules = settings.CustomRules;
                existingSettings.ModifiedDate = DateTime.UtcNow;
                
                settings = existingSettings;
            }

            await _context.SaveChangesAsync();
            return settings;
        }

        /// <summary>
        /// Create default league settings
        /// </summary>
        private LeagueSettings CreateDefaultSettings(Guid seasonId)
        {
            return new LeagueSettings
            {
                Id = Guid.NewGuid(),
                SeasonId = seasonId,
                HandicapMethod = HandicapCalculationMethod.WorldHandicapSystem,
                CoursePar = 36,
                CourseRating = 35.0m,
                SlopeRating = 113m,
                MaxRoundsForHandicap = 20,
                ScoringMethod = ScoringMethod.MatchPlay,
                PointsSystem = PointsSystem.HolePointsWithMatchBonus,
                HoleWinPoints = 2,
                HoleHalvePoints = 1,
                MatchWinBonus = 2,
                MatchTiePoints = 1,
                UseSessionHandicaps = true,
                AllowHandicapUpdates = true,
                CreatedDate = DateTime.UtcNow
            };
        }

        /// <summary>
        /// Get all league settings for administrative purposes
        /// </summary>
        public async Task<List<LeagueSettings>> GetAllLeagueSettingsAsync()
        {
            return await _context.LeagueSettings
                .Include(ls => ls.Season)
                .OrderByDescending(ls => ls.CreatedDate)
                .ToListAsync();
        }

        /// <summary>
        /// Reset league settings to defaults for a season
        /// </summary>
        public async Task<LeagueSettings> ResetToDefaultsAsync(Guid seasonId)
        {
            var existingSettings = await _context.LeagueSettings
                .FirstOrDefaultAsync(ls => ls.SeasonId == seasonId);

            if (existingSettings != null)
            {
                _context.LeagueSettings.Remove(existingSettings);
            }

            var defaultSettings = CreateDefaultSettings(seasonId);
            _context.LeagueSettings.Add(defaultSettings);
            await _context.SaveChangesAsync();

            return defaultSettings;
        }

        /// <summary>
        /// Delete league settings for a season
        /// </summary>
        public async Task<bool> DeleteLeagueSettingsAsync(Guid seasonId)
        {
            var settings = await _context.LeagueSettings
                .FirstOrDefaultAsync(ls => ls.SeasonId == seasonId);

            if (settings == null)
                return false;

            _context.LeagueSettings.Remove(settings);
            await _context.SaveChangesAsync();
            return true;
        }
    }
}

File: frontend/src/app/modules/settings/services/league-settings.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface LeagueSettings {
    id: string;
    seasonId: string;
    leagueName: string;
    handicapMethod: HandicapCalculationMethod;
    averageMethod: AverageCalculationMethod;
    legacyInitialWeight: number;
    coursePar: number;
    courseRating: number;
    slopeRating: number;
    maxRoundsForHandicap: number;
    scoringMethod: ScoringMethod;
    pointsSystem: PointsSystem;
    holeWinPoints: number;
    holeHalvePoints: number;
    matchWinBonus: number;
    matchTiePoints: number;
    useSessionHandicaps: boolean;
    allowHandicapUpdates: boolean;
    customRules?: string;
    createdDate: string;
    modifiedDate?: string;
}

export enum HandicapCalculationMethod {
    WorldHandicapSystem = 0,
    SimpleAverage = 1,
    LegacyLookupTable = 2
}

export enum AverageCalculationMethod {
    SimpleAverage = 0,
    LegacyWeightedAverage = 1
}

export enum ScoringMethod {
    MatchPlay = 0,
    StrokePlay = 1
}

export enum PointsSystem {
    HolePointsWithMatchBonus = 0,
    ScoreBasedPoints = 1,
    Custom = 2
}

export interface EnumOption {
    value: number;
    name: string;
}

@Injectable({
    providedIn: 'root'
})
export class LeagueSettingsService {
    private apiUrl = '/api/LeagueSettings';

    constructor(private http: HttpClient) { }

    getLeagueSettings(seasonId: string): Observable<LeagueSettings> {
        return this.http.get<LeagueSettings>(`${this.apiUrl}/season/${seasonId}`);
    }

    updateLeagueSettings(seasonId: string, request: any): Observable<LeagueSettings> {
        return this.http.put<LeagueSettings>(`${this.apiUrl}/season/${seasonId}`, request);
    }

    resetToDefaults(seasonId: string): Observable<LeagueSettings> {
        return this.http.post<LeagueSettings>(`${this.apiUrl}/season/${seasonId}/reset`, {});
    }

    getHandicapMethods(): Observable<EnumOption[]> {
        return this.http.get<EnumOption[]>(`${this.apiUrl}/enums/handicap-methods`);
    }

    getAverageMethods(): Observable<EnumOption[]> {
        return this.http.get<EnumOption[]>(`${this.apiUrl}/enums/average-methods`);
    }

    getScoringMethods(): Observable<EnumOption[]> {
        return this.http.get<EnumOption[]>(`${this.apiUrl}/enums/scoring-methods`);
    }

    getPointsSystems(): Observable<EnumOption[]> {
        return this.http.get<EnumOption[]>(`${this.apiUrl}/enums/points-systems`);
    }
}
